lib/inspec/base_cli.rb

```diff
@@ -71,24 +73,70 @@ def self.exec_options
         desc: 'Write out a lockfile based on this execution (unless one already exists)'
       option :backend_cache, type: :boolean,
         desc: 'Allow caching for backend command output.'
+      option :show_progress, type: :boolean,
+        desc: 'Show progress while executing tests.'
     end
 
     def self.default_options
       {
         exec: {
+          'reporter' => {
+            'cli' => nil,
+          },
+          'show_progress' => false,
           'color' => true,
           'create_lockfile' => true,
           'backend_cache' => false,
         },
       }
     end
 
+    def self.clean_reporters(opts)
+      # Parse report options from CLI
+      if opts['reporter'].is_a?(Array)
+        stdout = 0
+        reports = {}
+        opts['reporter'].each do |report|
+          k, v = report.split(':')
+          v = nil if v == '-'
+          reports[k] = v
+          stdout += 1 if v.nil?
+        end
+
+        raise ArgumentError, 'The option --reporter can only have a single report outputting to stdout.' if stdout > 1
+
+        opts['reporter'] = reports
```

Comment:
I propose a slightly different structure for internal reporter representation:
```
{
  "backend_cache": true,
  "reporter": {
    "junit": [
      {
        "file": "a.xml"
      }
    ],
    "json": [
      {
        "file": "a.json"
      }
    ]
  }
}
```
This allows us:
- more options for future reporter (e.g. the automate reporter that is coming with auth headers etc)
- we can parse the cli options into that structure and implement one validator method in the reporter implementation
- we do not need to distingish between cli and json config

We may enable yml as configuration in future. Above config would look like:
```
---
backend_cache: true
reporter:
  junit:
  - file: a.xml
  json:
  - file: a.json

```

Reply:
That's definitely more flexible. I don't know that it needs to be addressed in this PR since this an internal representation and nothing the user sees, and we can translate into whatever internal structure we need as our own needs and featureset changes... but it shouldn't be a big deal to do it in this PR if you feel strongly about it.